### PR TITLE
Fixing bugs of FE #2386

### PR DESCRIPTION
**Pull request by LianaHus**

### CODE REVIEW

File: src/app/files/file-explorer.js [hunk 1]
```diff
@@ -579,16 +584,15 @@ fileExplorer.prototype.copyFiles = function () {
...
 
-fileExplorer.prototype.createNewFile = function (parentFolder) {
+fileExplorer.prototype.createNewFile = function (parentFolder = 'browser') {
   let self = this
   modalDialogCustom.prompt('Create new file', 'File Name (e.g Untitled.sol)', 'Untitled.sol', (input) => {
-    helper.createNonClashingName(input, self.files, (error, newName) => {
-      if (error) return modalDialogCustom.alert('Failed to create file ' + newName + ' ' + error)
-      const currentPath = !parentFolder ? self._deps.fileManager.currentPath() : parentFolder
-      newName = currentPath ? currentPath + '/' + newName : self.files.type + '/' + newName
+    if (input === '') input = 'New file'
```
Comment (GrandSchtroumpf) [inline src/app/files/file-explorer.js]: I would rather do `if (input)` like that you check for null & undefined also.

File: src/app/files/file-explorer.js [hunk 2]
```diff
@@ -601,14 +605,24 @@ fileExplorer.prototype.createNewFile = function (parentFolder) {
 
 fileExplorer.prototype.createNewFolder = function (parentFolder) {
   let self = this
-  modalDialogCustom.prompt('Create new folder', '', '', (input) => {
+  modalDialogCustom.prompt('Create new folder', '', 'New folder', (input) => {
+    if (input === '') {
```
Comment (GrandSchtroumpf) [inline src/app/files/file-explorer.js]: Same as above

File: src/app/files/file-explorer.js [hunk 3]
```diff
@@ -601,14 +605,24 @@ fileExplorer.prototype.createNewFile = function (parentFolder) {
...
+      tooltip('Failed to create folder. The name can not be empty')
+      return false
```
Comment (GrandSchtroumpf) [inline src/app/files/file-explorer.js]: If you're using `return` to stop the process of the scope I would rath do `return tooltip(...)`. Returning `false` feels like it's used somewhere. If this is the case then forget this comment.

File: src/app/files/fileProvider.js [hunk 4]
```diff
@@ -131,24 +133,43 @@ class FileProvider {
...
-      return false
-    }
-    const stat = window.remixFileSystem.statSync(unprefixedpath)
-    try {
-      if (stat.isDirectory()) {
-        window.remixFileSystem.rmdirSync(unprefixedpath, console.log)
-      } else {
-        window.remixFileSystem.unlinkSync(unprefixedpath, console.log)
+    path = this.removePrefix(path)
+    if (window.remixFileSystem.existsSync(path)) {
```
Comment (GrandSchtroumpf) [inline src/app/files/fileProvider.js]: What ??? We inject the FileSystem as a global value in the file system o.O"

File: src/app/files/fileProvider.js [hunk 5]
```diff
@@ -131,24 +133,43 @@ class FileProvider {
...
+      const stat = window.remixFileSystem.statSync(path)
+      try {
+        if (!stat.isDirectory()) {
+          window.remixFileSystem.unlinkSync(path, console.log)
+          this.event.trigger('fileRemoved', [this._normalizePath(path)])
+          return true
+        } else {
+          let items = window.remixFileSystem.readdirSync(path)
```
Comment (GrandSchtroumpf) [inline src/app/files/fileProvider.js]: You're not reassigning it. So you can use `const`

File: src/app/files/fileProvider.js [hunk 6]
```diff
@@ -131,24 +133,43 @@ class FileProvider {
...
+          if (items.length !== 0) {
+            items.forEach((item, index) => {
+              let curPath = path + '/' + item
```
Comment (GrandSchtroumpf) [inline src/app/files/fileProvider.js]: Maybe : 
```javascript
const currPath = `${path}/${item}`
```

Comment (LianaHus) [reply src/app/files/fileProvider.js]: why it is better?

Comment (GrandSchtroumpf) [reply src/app/files/fileProvider.js]: You don't reassign it so you can you `const`. 
Personally I find the `${a}divder${b}` writing more readable than using concatenation (just suggesting here).

**Pull request merged**